src/PHPStan/ProxyQueryDynamicReturnTypeExtension.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\DoctrineMongoDBAdminBundle\PHPStan;
+
+use Doctrine\ODM\MongoDB\Query\Builder;
+use PHPStan\Broker\Broker;
+use PHPStan\Reflection\BrokerAwareExtension;
+use PHPStan\Reflection\ClassReflection;
+use PHPStan\Reflection\MethodReflection;
+use PHPStan\Reflection\MethodsClassReflectionExtension;
+use Sonata\AdminBundle\Datagrid\ProxyQueryInterface;
+use Sonata\DoctrineMongoDBAdminBundle\Datagrid\ProxyQuery;
+
+final class ProxyQueryDynamicReturnTypeExtension implements MethodsClassReflectionExtension, BrokerAwareExtension
```

Comment:
I just see this now, but is it a good idea having this class in src ?

Phpstan is a dev-dependency.

Reply:
nope, I was trying some things some time ago and I mixed up code, it was unintended.